.github/workflows/test.yml

```diff
@@ -0,0 +1,26 @@
+name: pymongo-auth-aws
+
+on: [push]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        python-version: ["2.7", "3.4", "3.5", "3.6", "3.7", "3.8", "3.9", "3.10", "pypy-2.7", "pypy-3.7"]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: install requests
+        run: |
+          python -m pip install requests==2.12	
+        if: ${{ matrix.python-version == 3.4 || matrix.python-version == 3.5}} 
```

Comment:
A comment that explains why this is needed would be nice.

Reply:
Done.

Reply:
What's special about 2.12? It looks like requests 2.21 was the last release to support Python 3.4 and requests 2.25 was the last release to support Python 3.5.

Reply:
Oh, there was nothing special about 2.12. It was just a lower version than what it normally was looking for, so I thought it would work. I will switch it to 2.21

Reply:
It looks like the actual issue is that the versions of requests  and botocore have conflicting urllib3 requirements:
```
ERROR: requests 2.21.0 has requirement urllib3<1.25,>=1.21.1, but you'll have urllib3 1.25.7 which is incompatible.
ERROR: botocore 1.19.63 has requirement urllib3<1.25.8,>=1.25.4; python_version == "3.4", but you'll have urllib3 1.24.3 which is incompatible.
```

It appear the latest change (requests==2.21) is failing too. I suspect that installing requests manually is no longer needed because we do `python -m pip install -e .`. Can you try removing the "install requests" section? If that doesn't work I'm fine with going back to 2.12.

Reply:
Looks like removing the install requests section still makes it fail. I am adding it back with version 2.12. 
```
Run python -m pip install -e .
DEPRECATION: Python 3.4 support has been deprecated. pip 19.1 will be the last one supporting it. Please upgrade your Python as Python 3.4 won't be maintained after March 2019 (cf PEP 429).
Obtaining file:///home/runner/work/pymongo-auth-aws/pymongo-auth-aws
Collecting requests<3.0.0 (from pymongo-auth-aws==1.0.3.dev0)
  Downloading https://files.pythonhosted.org/packages/7d/e3/20f3d364d6c8e5d2353c72a67778eb189176f08e873c9900e10c0287b84b/requests-2.21.0-py2.py3-none-any.whl (57kB)
Collecting botocore (from pymongo-auth-aws==1.0.3.dev0)
  Downloading https://files.pythonhosted.org/packages/2c/05/0a955f0c92bec7da076fbbc73926dfb13fab8e2b88de7f8eb17c443f28f0/botocore-1.19.63-py2.py3-none-any.whl (7.2MB)
Collecting urllib3<1.25,>=1.21.1 (from requests<3.0.0->pymongo-auth-aws==1.0.3.dev0)
  Downloading https://files.pythonhosted.org/packages/01/11/525b02e4acc0c747de8b6ccdab376331597c569c42ea66ab0a1dbd36eca2/urllib3-1.24.3-py2.py3-none-any.whl (118kB)
Collecting chardet<3.1.0,>=3.0.2 (from requests<3.0.0->pymongo-auth-aws==1.0.3.dev0)
  Downloading https://files.pythonhosted.org/packages/bc/a9/01ffebfb562e4274b6487b4bb1ddec7ca55ec7510b22e4c51f14098443b8/chardet-3.0.4-py2.py3-none-any.whl (133kB)
Collecting idna<2.9,>=2.5 (from requests<3.0.0->pymongo-auth-aws==1.0.3.dev0)
  Downloading https://files.pythonhosted.org/packages/14/2c/cd551d81dbe15200be1cf41cd03869a46fe7226e7450af7a6545bfc474c9/idna-2.8-py2.py3-none-any.whl (58kB)
Collecting certifi>=2017.4.17 (from requests<3.0.0->pymongo-auth-aws==1.0.3.dev0)
  Downloading https://files.pythonhosted.org/packages/37/45/946c02767aabb873146011e665728b680884cd8fe70dde973c640e45b775/certifi-2021.10.8-py2.py3-none-any.whl (149kB)
Collecting jmespath<1.0.0,>=0.7.1 (from botocore->pymongo-auth-aws==1.0.3.dev0)
  Downloading https://files.pythonhosted.org/packages/07/cb/5f001272b6faeb23c1c9e0acc04d48eaaf5c862c17709d20e3469c6e0139/jmespath-0.10.0-py2.py3-none-any.whl
Collecting python-dateutil<3.0.0,>=2.1 (from botocore->pymongo-auth-aws==1.0.3.dev0)
  Downloading https://files.pythonhosted.org/packages/36/7a/87837f39d0296e723bb9b62bbb257d0355c7f6128853c78955f57342a56d/python_dateutil-2.8.2-py2.py3-none-any.whl (247kB)
Collecting six>=1.5 (from python-dateutil<3.0.0,>=2.1->botocore->pymongo-auth-aws==1.0.3.dev0)
  Downloading https://files.pythonhosted.org/packages/d9/5a/e7c31adbe875f2abbb91bd84cf2dc52d792b5a01506781dbcf25c91daf11/six-1.16.0-py2.py3-none-any.whl
ERROR: botocore 1.19.63 has requirement urllib3<1.25.8,>=1.25.4; python_version == "3.4", but you'll have urllib3 1.24.3 which is incompatible.
Installing collected packages: urllib3, chardet, idna, certifi, requests, jmespath, six, python-dateutil, botocore, pymongo-auth-aws
  Running setup.py develop for pymongo-auth-aws
Successfully installed botocore-1.19.63 certifi-2021.10.8 chardet-3.0.4 idna-2.8 jmespath-0.10.0 pymongo-auth-aws python-dateutil-2.8.2 requests-2.21.0 six-1.16.0 urllib3-1.24.3
WARNING: You are using pip version 19.1.1, however version 21.3.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
running test
Searching for urllib3<1.25.8,>=1.25.4; python_version == "3.4"
Reading https://pypi.python.org/simple/urllib3/
Couldn't retrieve index page for 'urllib3'
Scanning index of all packages (this may take a while)
Reading https://pypi.python.org/simple/
No local packages or working download links found for urllib3<1.25.8,>=1.25.4; python_version == "3.4"
error: Could not find suitable distribution for Requirement.parse('urllib3<1.25.8,>=1.25.4; python_version == "3.4"')
Error: Process completed with exit code 1.```